unit/src/FunctionTest.C

```diff
@@ -25,11 +25,11 @@ TEST_F(FunctionTest, GetScalarFunctionCoefficient)
   func_params1.set<std::string>("expression") = "1.";
   _mfem_problem->addFunction("ParsedFunction", "coef1", func_params1);
   _mfem_problem->getFunction("coef1").initialSetup();
-  std::shared_ptr<mfem::Coefficient> coef = _mfem_problem->getScalarFunctionCoefficient("coef1");
-  EXPECT_EQ(coef->Eval(fe_transform, point), 1);
+  mfem::Coefficient & coef(_mfem_problem->getCoefficients().getScalarCoefficient("coef1"));
+  EXPECT_EQ(coef.Eval(fe_transform, point), 1);
 
-  EXPECT_THROW(_mfem_problem->getVectorFunctionCoefficient("coef1"), std::runtime_error);
-  EXPECT_THROW(_mfem_problem->getScalarFunctionCoefficient("coef2"), std::runtime_error);
+  EXPECT_THROW(_mfem_problem->getCoefficients().getVectorCoefficient("coef1"), MooseException);
+  EXPECT_THROW(_mfem_problem->getCoefficients().getScalarCoefficient("coef2"), MooseException);
```

Comment:
Is there a good way to catch that `mooseError()` has been called, if we switch to using that instead of `MooseException`?

Reply:
Since I just changed the throw type from `mooseError` to be `MooseException` during unit testing, this code doesn't need any changes ... we'll still catch `MooseException`